folio-spring-i18n/src/main/java/org/folio/spring/i18n/model/TranslationMap.java

```diff
@@ -0,0 +1,190 @@
+package org.folio.spring.i18n.model;
+
+import com.ibm.icu.text.MessageFormat;
+import jakarta.annotation.Nullable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * This object represents the relationship between a locale and a
+ * {@link TranslationFile TranslationFile}.  The quality of this relationship is determined by how
+ * well the file's locale matches up with the one in use.  This object will also contain fallbacks,
+ * such as alternative languages for the user, or the server's default locale, to provide more
+ * seamless support for missing translations.
+ */
+@Log4j2
+@ToString
+@EqualsAndHashCode
+public final class TranslationMap {
+
+  @Getter
+  private final Locale locale;
+
+  @Getter
+  private final TranslationMatchQuality quality;
+
+  @Getter
+  private final TranslationFile file;
+
+  private final Map<String, String> patterns;
+
+  @Getter
+  @Nullable
+  private final TranslationMap fallback;
+
+  /**
+   * Create a TranslationMap for a given locale based off of a file with contents and a given fallback.
+   *
+   * @param locale the associated locale
+   * @param file the {@link TranslationFile TranslationFile} to infer quality from
+   * @param patterns the patterns to use
+   * @param fallback the TranslationMap to search if a given translation cannot be found
+   */
+  private TranslationMap(
+    Locale locale,
+    TranslationFile file,
+    Map<String, String> patterns,
+    @Nullable TranslationMap fallback
+  ) {
```

Comment:
Are there any pros of such formatting? For me, it's less readable and just adding additional rows to the file

Reply:
It could be combined into one line, which would make a 130 character long line that goes off the screen.
This could be split into two lines or something, but it still seems a bit too packed together. IMO more space is better for a case like this, where you want to be able to easily scan and find the method parameters.
<img width="975" alt="image" src="https://github.com/folio-org/folio-spring-support/assets/8005215/70f7d14f-5b83-46bf-80ff-dbe0b919d693">